app/src/main/java/com/sapuseven/untis/activities/MainActivity.kt

```diff
@@ -817,6 +819,12 @@ class MainAppState @OptIn(ExperimentalMaterial3Api::class) constructor(
 		items: List<TimegridItem>
 	): List<TimegridItem> {
 		val newItems = mergeItems(items.mapNotNull { item ->
+			if (displayedElement.value?.type?.equals(TimetableDatabaseInterface.Type.SUBJECT.name) == false
+				&& item.periodData.subjects.isNotEmpty()
+				&& decodeMultipleStoredTimetableValues(preferences.timetableHiddenElements.getValue())?.contains(item.periodData.subjects.first()) == true
+			) {
+				return@mapNotNull null
+			}
 			if (item.periodData.isCancelled() && preferences.timetableHideCancelled.getValue())
 				return@mapNotNull null
```

Comment:
Same case as in `NotificationSetupWorker`.
```suggestion
		val hiddenSubjects = decodeMultipleStoredTimetableValues(preferences.timetableHiddenElements.getValue()).orEmpty()

		val newItems = mergeItems(items.mapNotNull { item ->
			if (item.periodData.isCancelled() && preferences.timetableHideCancelled.getValue())
				return@mapNotNull null

			if (displayedElement.value?.type?.equals(TimetableDatabaseInterface.Type.SUBJECT.name) == false
				&& item.periodData.subjects.isNotEmpty()
				&& hiddenSubjects.containsAll(item.periodData.subjects)
			) {
				return@mapNotNull null
			}
```